src/linux/timer.c

```diff
@@ -151,7 +151,10 @@ evfilt_timer_knote_create(struct filter *filt, struct knote *kn)
     }
 
     memset(&ev, 0, sizeof(ev));
-    ev.events = EPOLLIN;
+    ev.events = EPOLLIN | EPOLLET;
+    if (kn->kev.flags & EV_ONESHOT || kn->kev.flags & EV_DISPATCH)
```

Comment:
`if ((kn->kev.flags & EV_ONESHOT) || (kn->kev.flags & EV_DISPATCH))`

is the desired logic.